RobotController/robot_brain.py

```diff
@@ -61,23 +65,35 @@ def _check_next_evaluation(self, force=False):
         Check if is a moment for a new evaluation and starts a new one
         :param force: forces the new evaluation to start
         """
+        if self.mating_client and self.mating_client.availability:
+            self.mating_client.server.join()
+            self.mating_client.client.join()
+            self.mating_client = None
```

Comment:
You tell me XD 
How much time does it take to do both joins? (how much does it take for the two threads to finish their work)